src/LiveComponent/src/Util/QueryStringPropsExtractor.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent\Util;
+
+use Symfony\Component\HttpFoundation\HeaderUtils;
+use Symfony\Component\PropertyInfo\Type;
+use Symfony\UX\LiveComponent\Metadata\LiveComponentMetadata;
+
+/**
+ * @author Nicolas Rigaud <squrious@protonmail.com>
+ *
+ * @experimental
+ *
+ * @internal
+ */
+class QueryStringPropsExtractor
+{
+    public function extract(?string $queryString, LiveComponentMetadata $metadata): array
+    {
+        if (empty($queryString)) {
+            return [];
+        }
+
+        $query = HeaderUtils::parseQuery($queryString);
+
+        $data = [];
+
+        foreach ($metadata->getAllLivePropsMetadata() as $livePropMetadata) {
+            $queryStringBinding = $livePropMetadata->getQueryStringMapping();
+            foreach ($queryStringBinding['parameters'] ?? [] as $parameterName => $paramConfig) {
```

Comment:
I chose to always return an array from `$livePropMetadata->getQueryStringMapping()`, possibly populated with configuration values, such as `parameters`. So all unbound props have an empty array here. Maybe returning `false` instead of the empty array could be better?

Reply:
I see - I'm ok with it.